RubberduckTests/Inspections/EmptyIfBlockInspectionTests.cs

```diff
@@ -0,0 +1,958 @@
+            var state = MockParser.CreateAndParse(vbe.Object);
+
+            var inspection = new EmptyIfBlockInspection(state);
+            var inspector = InspectionsHelper.GetInspector(inspection);
+            var inspectionResults = inspector.FindIssuesAsync(state, CancellationToken.None).Result;
+
+            Assert.AreEqual(1, inspectionResults.Count());
+        }
+
+        [TestMethod]
+        [TestCategory("Inspections")]
+        public void EmptyIfBlock_FiresOnEmptyIfBlock_HasQuoteComment()
+        {
+            const string inputCode =
+@"Sub Foo()
+    If True Then
+        ' Im a comment
+    End If
+End Sub";
+
+            IVBComponent component;
+            var vbe = MockVbeBuilder.BuildFromSingleStandardModule(inputCode, out component);
+            var state = MockParser.CreateAndParse(vbe.Object);
+
+            var inspection = new EmptyIfBlockInspection(state);
+            var inspector = InspectionsHelper.GetInspector(inspection);
+            var inspectionResults = inspector.FindIssuesAsync(state, CancellationToken.None).Result;
+
+            Assert.AreEqual(1, inspectionResults.Count());
+        }
+
+        [TestMethod]
+        [TestCategory("Inspections")]
+        public void EmptyIfBlock_FiresOnEmptyIfBlock_HasRemComment()
+        {
+            const string inputCode =
+@"Sub Foo()
+    If True Then
+        Rem Im a comment
+    End If
+End Sub";
+
+            IVBComponent component;
+            var vbe = MockVbeBuilder.BuildFromSingleStandardModule(inputCode, out component);
+            var state = MockParser.CreateAndParse(vbe.Object);
+
+            var inspection = new EmptyIfBlockInspection(state);
+            var inspector = InspectionsHelper.GetInspector(inspection);
+            var inspectionResults = inspector.FindIssuesAsync(state, CancellationToken.None).Result;
+
+            Assert.AreEqual(1, inspectionResults.Count());
+        }
+
+        [TestMethod]
+        [TestCategory("Inspections")]
+        public void EmptyIfBlock_FiresOnEmptyIfBlock_HasWhitespace()
+        {
+            const string inputCode =
+@"Sub Foo()
+    If True Then
+
+    	
+    End If
+End Sub";
+
+            IVBComponent component;
+            var vbe = MockVbeBuilder.BuildFromSingleStandardModule(inputCode, out component);
+            var state = MockParser.CreateAndParse(vbe.Object);
+
+            var inspection = new EmptyIfBlockInspection(state);
+            var inspector = InspectionsHelper.GetInspector(inspection);
+            var inspectionResults = inspector.FindIssuesAsync(state, CancellationToken.None).Result;
+
+            Assert.AreEqual(1, inspectionResults.Count());
+        }
+
+        [TestMethod]
+        [TestCategory("Inspections")]
+        public void EmptyIfBlock_IfBlockHasExecutableStatement()
+        {
+            const string inputCode =
+@"Sub Foo()
+    If True Then
+        Dim d
```

Comment:
`Dim` is not an executable statement, and it's scoped to the smallest possible scope: procedure-level. This case should be considered an empty `If` block (the fix would move the declaration outside the block and then delete the whole thing I guess)